lib/matplotlib/cbook.py

```diff
@@ -2490,6 +2490,34 @@ def pts_to_midstep(x, *args):
                    'step-post': pts_to_poststep,
                    'step-mid': pts_to_midstep}
 
+
+def get_index_y(y):
+    """
+    A helper function to get the index of an input to plot
+    against if x values are not explicitly given.
+
+    Tries to get `y.index` (works if this is a pd.Series), if that
+    fails, return np.arange(y.shape[0]).
+
+    This will be extended in the future to deal with more types of
+    labeled data.
+
+    Parameters
+    ----------
+    y : scalar or array-like
+        The proposed y-value
+
+    Returns
+    -------
+    x, y : ndarray
+       The x and y values to plot.
+    """
+    try:
+        return y.index.values, y.values
+    except AttributeError:
+        y = np.atleast_1d(y)
+        return np.arange(y.shape[0], dtype=float), y
+
```

Comment:
Calling `series.index.values` on a Series with a MultiIndex will return a 1d array of tuples.